components/experimental/src/dimension/provider/extended_currency.rs

```diff
@@ -0,0 +1,85 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use icu_provider::prelude::*;
+use zerovec::ZeroMap;
+
+#[cfg(feature = "compiled_data")]
+/// Baked data
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. In particular, the `DataProvider` implementations are only
+/// guaranteed to match with this version's `*_unstable` providers. Use with caution.
+/// </div>
+pub use crate::provider::Baked;
+
+/// Currency Extended V1 data struct.
+#[icu_provider::data_struct(marker(CurrencyExtendedDataV1Marker, "currency/extended@1"))]
+#[derive(Debug, Clone, Default, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_experimental::dimension::provider::extended_currency)
+)]
+#[yoke(prove_covariance_manually)]
+pub struct CurrencyExtendedDataV1<'data> {
+    // TODO: Implement currency pattern selection logic to choose between standard or standard next to text pattern.
+    /// Contains the localized display names for a currency based on plural rules.
+    /// For instance, in the "en" locale for the "USD" currency:
+    ///     - "US Dollars" when count is `zero`,
+    ///     - "US Dollar" when count is `one`,
+    ///     ... etc.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub display_names: ZeroMap<'data, Count, str>,
```

Comment:
Issue: As described in [the spec](https://unicode.org/reports/tr35/tr35-numbers.html#Currencies), you also need to include `unitPattern` along with the display names. Consider adding it to currency essentials.

Reply:
The patterns are already in currency essentials. I just need to add the selector in the next PR to choose the standard or standard-alphaNextToNumber

Reply:
I think unitPattern is different than standard and standard-alphaNextToNumber. It is a third pattern. 

But you can add it in the next PR. This one is fine.